airgap/v2/cmd/client/download/download.go

```diff
@@ -16,100 +16,94 @@ package download
+	outputDirectory  string
+	fileListPath     string
+	deleteOnDownload bool
+	conn             *grpc.ClientConn
+	client           fileretreiver.FileRetreiverClient
+	log              logr.Logger
 }
 
-var (
-	dc  DownloadConfig
-	log logr.Logger
-)
+var dc DownloadConfig
 
 // DownloadCmd represents the download command
 var DownloadCmd = &cobra.Command{
 	Use:   "download",
 	Short: "Download files from the airgap service",
 	Long:  `An external configuration file containing connection details are expected`,
+	Example: `
+    # Download file based using file name 
+    client download --file-name file_name --output-directory /path/to/output/dir --config /path/to/config.yaml
+    
+    # Download file based using file id 
+    client download --file-id file_id --output-directory /path/to/output/dir --config /path/to/config.yaml
+    
+    # Download files in batch using csv file 
+    client download --file-list-path file_name --output-directory /path/to/output/dir --config /path/to/config.yaml
+    
+    # Mark file for deletion on download 
+    client download -D --file-name file_name --output-directory /path/to/output/dir --config /path/to/config.yaml`,
 	RunE: func(cmd *cobra.Command, args []string) error {
+		//Initialize logger
+		initLog()
 		// Initialize client
 		err := dc.initializeDownloadClient()
 		if err != nil {
 			return err
 		}
 		defer dc.closeConnection()
 
-		// Download file and return error if any
-		return dc.downloadFile()
+		// If file list path is specified, perform batch download
+		if len(strings.TrimSpace(dc.fileListPath)) != 0 {
+			return dc.batchDownload()
+		}
+
+		// If file name/identifier is specified, download the single file
+		return dc.downloadFile(dc.fileName, dc.fileId)
 	},
 }
 
 func init() {
-	initLog()
 	DownloadCmd.Flags().StringVarP(&dc.fileName, "file-name", "n", "", "Name of the file to be downloaded")
 	DownloadCmd.Flags().StringVarP(&dc.fileId, "file-id", "i", "", "Id of the file to be downloaded")
-	DownloadCmd.Flags().StringVarP(&dc.outputDirectory, "output-directory", "o", "", "Path to download the file ")
+	DownloadCmd.Flags().StringVarP(&dc.outputDirectory, "output-directory", "o", "", "Path to download the file")
+	DownloadCmd.Flags().StringVarP(&dc.fileListPath, "file-list-path", "f", "", "Fully qualified path to file containing list of names/identifiers")
+	DownloadCmd.Flags().BoolVarP(&dc.deleteOnDownload, "delete-on-download", "D", false, "Mark file for deletion")
 	DownloadCmd.MarkFlagRequired("output-directory")
 }
 
+// initLog initializes logger
 func initLog() {
-	zapLog, err := zap.NewDevelopment()
+	var err error
+	dc.log, err = util.InitLog()
 	if err != nil {
-		panic(fmt.Sprintf("Failed to initialize zapr, due to error: %v", err))
+		panic(err)
 	}
-	log = zapr.NewLogger(zapLog)
 }
 
 // initializeDownloadClient initializes the file retriever client based on provided configuration parameters
 func (dc *DownloadConfig) initializeDownloadClient() error {
```

Comment:
Let's move this into a Provider.

Example:
```go
func ProvideDownloadClient( log, ...args ) (DownloadClient, error)
```

Reply:
Providers are basically a Java New that can fail. You would use them in injectors.